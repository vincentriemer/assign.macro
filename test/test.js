const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import assign from "../dist/assign.macro";

      const test = {};

      assign(test, { 
        babel: "macros",
        are: "dope" 
      });
    `,
    `
      import assign from "../dist/assign.macro";

      const test = {};
      const foo = "bar";

      assign(test, { foo });
    `,
    `
      import assign from "../dist/assign.macro";

      const test = {};
      const testProp1 = "bar";

      assign(test, { 
        testProp1,
        testProp2: () => {
          console.log("foo");
        },
        testProp3: 5,
        testProp4: ["foo", "bar"],
        testProp5: { foo: "bar" }
      });
    `
  ]
});
