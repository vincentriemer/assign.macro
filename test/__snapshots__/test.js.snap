// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import assign from \\"../dist/assign.macro\\";

const test = {};

assign(test, { 
  babel: \\"macros\\",
  are: \\"dope\\" 
});

      ↓ ↓ ↓ ↓ ↓ ↓

const test = {};

test.babel = \\"macros\\";
test.are = \\"dope\\";
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import assign from \\"../dist/assign.macro\\";

const test = {};
const foo = \\"bar\\";

assign(test, { foo });

      ↓ ↓ ↓ ↓ ↓ ↓

const test = {};
const foo = \\"bar\\";

test.foo = foo;
"
`;

exports[`macros 3. macros: 3. macros 1`] = `
"
import assign from \\"../dist/assign.macro\\";

const test = {};
const testProp1 = \\"bar\\";

assign(test, { 
  testProp1,
  testProp2: () => {
    console.log(\\"foo\\");
  },
  testProp3: 5,
  testProp4: [\\"foo\\", \\"bar\\"],
  testProp5: { foo: \\"bar\\" }
});

      ↓ ↓ ↓ ↓ ↓ ↓

const test = {};
const testProp1 = \\"bar\\";

test.testProp1 = testProp1;

test.testProp2 = () => {
  console.log(\\"foo\\");
};

test.testProp3 = 5;
test.testProp4 = [\\"foo\\", \\"bar\\"];
test.testProp5 = { foo: \\"bar\\" };
"
`;
